libtcmu.c

```diff
@@ -138,6 +138,12 @@ lookup_dev_by_name(struct tcmulib_context *ctx, char *dev_name, int *index)
 	darray_foreach(dev_ptr, ctx->devices) {
 		dev = *dev_ptr;
 		size_t len = strnlen(dev->dev_name, sizeof(dev->dev_name));
+		if (len == sizeof(dev->dev_name)) {
+			char err_dev[256];
```

Comment:
Where did 256 come from?

I think the max len it could be is going to be strlen(uio) + max number of chars to represent an int which I think is 10 + plus null char.

Maybe add a define for the magic number we use, so we can change it later.

I am not in favor of adding extra checks because we think there is a bug and we can't root cause it. The input we get at the entry points should be checked and once it passes, the rest of the code does not need to double check it. It will get crazy with tests if we have to double check everything.

For example, I think you should add checks in open_devices that the sysfs name we are using is the format we expect, and in handle_netlink we should verify we got what we expected. Is nla_get_u32 using unsigned but other places are using a singed int an issue.